sled-agent/src/support_bundle/storage.rs

```diff
@@ -436,7 +482,10 @@ impl<'a> SupportBundleManager<'a> {
             // The dataset for a support bundle exists.
             let support_bundle_path = dataset
                 .name
-                .mountpoint(&self.storage.zpool_mountpoint_root())
+                .ensure_mounted_and_get_mountpoint(
```

Comment:
Every spot where we read the mountpoint of the support bundle dataset: Instead, *ensure it's actually mounted* first.

This is tested in `listing_bundles_mounts_them`